Sources/Sharing/Internal/Reference.swift

```diff
@@ -454,85 +449,6 @@ extension _PersistentReference: MutableReference, Equatable where Key: SharedKey
   }
 }
 
-final class _ManagedReference<Key: SharedReaderKey>: Reference, Observable {
```

Comment:
@pyrtsa Thanks for the PR and for looking into this! It's been a little while since we worked on this, but I believe our worry at the time was that two shared references could compete and initialize the same persistent reference at the same time, and that they would hold unique references while only a single one raced to live in the global persistent references store.

When looking into this did you conclude that this is not possible due to the existing locking?

I also wonder if it's possible to cook up a test case that deadlocks on `main` and is fixed in this branch?

Reply:
I wrote it so that while two or more threads can indeed _start_ setting up the shared reference, only one of them will succeed (while holding the lock another time), and all others will simply drop their half-initialised reference and grab the one already there.

Reply:
I need to think a bit on the failing/fixing unit test case, maybe it's possible with some artificial delays. 🤔

Reply:
There's a unit test now.

I tried to make the CI run the commit with just the failing test on top of `main`, but [GHA got stuck somehow](https://github.com/pointfreeco/swift-sharing/actions/runs/14861719272).

Anyway, you can check out d188bfb and run tests manually to see that it indeed deadlocks.